caravel/assets/javascripts/explorev2/components/ChartControl.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import Select from 'react-select';
+import { bindActionCreators } from 'redux';
+import * as actions from '../actions/exploreActions';
+import { connect } from 'react-redux';
+import { VIZ_TYPES } from '../../../utils/common';
+
+class ChartControl extends React.Component {
+  componentWillMount() {
+    if (this.props.datasourceId) {
+      this.props.actions.setFormOpts(this.props.datasourceId, this.props.datasourceType);
```

Comment:
once we import actions above we can use them like this:
`setFormOpts(this.props.datasourceId, this.props.datasourceType);`